File: exercise/java/day23/src/main/java/app/Main.java
package app;

import core.control.ControlSystem;
import core.control.SleighNotStartedException;
import core.control.ReindeersNeedRestException;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        ControlSystem controlSystem = new ControlSystem();
        controlSystem.startSystem();

        Scanner scanner = new Scanner(System.in);
        boolean keepRunning = true;

        while (keepRunning) {
            System.out.println("Enter a command (ascend (a), descend (d), park (p), or quit (q)): ");
            String command = scanner.nextLine();

            switch (command) {
                case "ascend": case "a":
                    try {
                        controlSystem.ascend();
                    } catch (ReindeersNeedRestException | SleighNotStartedException e) {
                        System.out.println(e.getMessage());
                    }
                    break;
                case "descend":  case "d":
                    try {
                        controlSystem.descend();
                    } catch (SleighNotStartedException e) {
                        System.out.println(e.getMessage());
                    }
                    break;
                case "park": case "p":
                    try {
                        controlSystem.park();
                    } catch (SleighNotStartedException e) {
                        System.out.println(e.getMessage());
                    }
                    break;
                case "quit": case "q":
                    keepRunning = false;
                    break;
                default:
                    System.out.println("Invalid command. Please try again.");
            }
        }

        scanner.close();
        controlSystem.stopSystem();
    }
}

File: exercise/kotlin/day23/src/main/kotlin/app/Main.kt
package app

import core.control.ControlSystem
import core.control.ReindeersNeedRestException
import core.control.SleighNotStartedException
import java.util.*

fun main() {
    val controlSystem = ControlSystem()
    controlSystem.startSystem()

    val scanner = Scanner(System.`in`)
    var keepRunning = true

    while (keepRunning) {
        println("Enter a command (ascend (a), descend (d), park (p), or quit (q)): ")
        val command = scanner.nextLine()

        when (command) {
            "ascend", "a" -> try {
                controlSystem.ascend()
            } catch (e: ReindeersNeedRestException) {
                println(e.message)
            } catch (e: SleighNotStartedException) {
                println(e.message)
            }

            "descend", "d" -> try {
                controlSystem.descend()
            } catch (e: SleighNotStartedException) {
                println(e.message)
            }

            "park", "p" -> try {
                controlSystem.park()
            } catch (e: SleighNotStartedException) {
                println(e.message)
            }

            "quit", "q" -> keepRunning = false
            else -> println("Invalid command. Please try again.")
        }
    }

    scanner.close()
    controlSystem.stopSystem()
}